jina/peapods/replicas/__init__.py

```diff
@@ -0,0 +1,345 @@
+__copyright__ = "Copyright (c) 2020 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+import argparse
+from argparse import Namespace
+from contextlib import ExitStack
+from typing import Optional, Dict, List, Union, Set
+
+from .helper import (
+    _set_peas_args,
+    _set_after_to_pass,
+    _copy_to_head_args,
+    _copy_to_tail_args,
+    _fill_in_host,
+)
+from ..peas import BasePea
+from ...enums import *
+
+
+class BaseReplica(ExitStack):
```

Comment:
In order to have Replication transparent to the Flow, a Replica should fullfill Pod interface, thus being a Pod itself I think